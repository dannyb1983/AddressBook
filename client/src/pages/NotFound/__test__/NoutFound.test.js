//modules
import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
//components
import NotFound from "../NotFound";

describe("NotFound (Unit)", () => {
  afterEach(() => cleanup());
  const renderNotFound = () =>
    render(
      <MemoryRouter initialEntries={["/blah"]}>
        <NotFound />
      </MemoryRouter>
    );
  const renderNotFoundBadAddress = () =>
    render(
      <MemoryRouter initialEntries={["/addcontacts"]}>
        <NotFound />
      </MemoryRouter>
    );

  describe("render NotFound()", () => {
    const { asFragment, container } = renderNotFound();

    it("NotFound render should match snapshot", () => {
      renderNotFound();
      expect(asFragment()).toMatchSnapshot();
    });

    it("NotFound message should render at an improper page address", () => {
      renderNotFound();
      expect(screen.getByText("Oops! Page Not Found")).toBeInTheDocument();
    });
  });
});
