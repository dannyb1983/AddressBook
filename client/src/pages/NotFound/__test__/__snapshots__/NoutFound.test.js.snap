// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound (Unit) render NotFound() NotFound render should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Oops! Page Not Found
    </h1>
  </div>
</DocumentFragment>
`;
